File: bench/data-structures-vector/1.cpp
#include <iostream>
#include <vector>
#include <chrono>

using namespace std;

int main(int argc, char const *argv[])
{
	vector<int> vec;

	int n = 100000000;
	if (argc == 2)
	{
		n = atoi(argv[1]);
	}

	auto start = chrono::steady_clock::now();

	for (int i = 0; i < n; i++)
	{
		vec.push_back(42);
	}

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}

File: bench/data-structures-vector/1.go
package main

import (
	"os"
	"strconv"
	"fmt"
	"time"
)

func main() {
	var n = 100000000
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.Atoi(args[0])
	}

	var start = time.Now()

	var arr = make([]int, 0, n)

	for i := 0; i < n; i++ {
		arr = append(arr, 42)
	}

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}
